tlsr9/ble/vendor/controller/b9x_bt_flash.c

```diff
@@ -27,35 +27,46 @@
 #include "flash.h"
 #include "string.h"
 #include <zephyr/storage/flash_map.h>
+#include <zephyr/drivers/flash.h>
+
+#if CONFIG_B9X_BLE_CTRL_MAC_FLASH
+static const struct device *flash_device = DEVICE_DT_GET_OR_NULL(DT_CHOSEN(zephyr_flash_controller));
+#endif
 
 /**
  * @brief		This function is used to initialize the MAC address
  * @param[in]	bt_mac - BT MAC address
- * @return      none
+ * @return      Status - 0: command succeeded
  */
-_attribute_no_inline_ void b9x_bt_blc_mac_init(uint8_t *bt_mac)
+_attribute_no_inline_ int b9x_bt_blc_mac_init(uint8_t *bt_mac)
 {
+	int err = 0;
+#if CONFIG_B9X_BLE_CTRL_MAC_FLASH
 	uint8_t dummy_mac[BLE_ADDR_LEN] = {0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF};
-
-	flash_read_page(FIXED_PARTITION_OFFSET(vendor_partition) +
-	B9X_BT_MAC_ADDR_OFFSET, BLE_ADDR_LEN, bt_mac);
-	if (!memcmp(bt_mac, dummy_mac, BLE_ADDR_LEN))
-    {
-		generateRandomNum(BLE_ADDR_LEN, bt_mac);
+	err = flash_read(flash_device, FIXED_PARTITION_OFFSET(vendor_partition) + B9X_BT_MAC_ADDR_OFFSET, bt_mac, BLE_ADDR_LEN);
+	if (err < 0)
+	{
+		return err;
+	}
+	if (memcmp(bt_mac, dummy_mac, BLE_ADDR_LEN))
```

Comment:
in the function _int b9x_bt_blc_init(void *prx, void *ptx_
we have
```
u8 ble_mac[BLE_ADDR_LEN];
b9x_bt_blc_mac_init(ble_mac);
```
semms like comparison of flash content with trash...